### PR TITLE
Kill some GH issues and all Redux anti-patterns 🔫

### PR DESCRIPTION
**Pull request by nop33**
@mvaivre I'll assign it for review tomorrow, I don't have the time to go through the PR today! But these should be all the changes I wanted to make to remove all anti-patterns that I became aware about (and also close some more GH issues)

### CODE REVIEW

File: src/utils/addresses.ts [hunk 1]
```diff
@@ -86,3 +88,21 @@ export const initializeAddressDiscoveryGroupsData = (addresses: Address[]): Addr
 
   return groupsData
 }
+
+export const getAvailableBalance = (address: Address): bigint => BigInt(address.balance) - BigInt(address.lockedBalance)
```
Comment (mvaivre) [inline src/utils/addresses.ts:92]: Nit: `getAddressAvailableBalance`?

**Review by mvaivre — COMMENTED** [review]
That's mmmmh, super, super nice ✨ 👏

File: src/screens/SendScreen.tsx [hunk 2]
```diff
@@ -117,9 +107,27 @@ const SendScreen = ({
...
+  const [unsignedTxId, setUnsignedTxId] = useState('')
+  const [unsignedTransaction, setUnsignedTransaction] = useState('')
+  const [isConsolidateUTXOsModalVisible, setIsConsolidateUTXOsModalVisible] = useState(false)
+  const [consolidationRequired, setConsolidationRequired] = useState(false)
+  const [isSweeping, setIsSweeping] = useState(false)
+  const [sweepUnsignedTxs, setSweepUnsignedTxs] = useState<SweepAddressTransaction[]>([])
+  const [isAuthenticationModalVisible, setIsAuthenticationModalVisible] = useState(false)
+  const [txStep, setTxStep] = useState<TxStep>('build')
+  const [isBuildingTx, setIsBuildingTx] = useState(false)
+  const [isSendingTx, setIsSendingTx] = useState(false)
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/screens/SendScreen.tsx:123]: I knew you'd say sth about this ;) 

It's a good idea to group things that are related. If it was a form, for example, I'd do that. But I don't see how many of those are... I'll see what I can do, but I am not a huge fan of the proposal.

File: src/components/inputs/PinCodeInput.tsx [hunk 3]
```diff
@@ -16,40 +16,37 @@ You should have received a copy of the GNU Lesser General Public License
...
 import { StyleProp, View, ViewStyle } from 'react-native'
 import styled from 'styled-components/native'
 
 import NumberKeyboard, { NumberKeyboardKey } from '../keyboard/NumberKeyboard'
 
 interface PinInputProps {
   pinLength: number
-  value: string
-  onPinChange: (value: string) => void
+  onPinChange: (value: string) => Promise<boolean> | boolean
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/components/inputs/PinCodeInput.tsx:27]: Totally agreed, thanks for spotting this

File: src/screens/SendScreen.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nop33) [reply src/screens/SendScreen.tsx]: I did a major refactoring of the send screen, made the code much dryer and cleaner! I grouped a new `useState`s and removed a substatial amount of them, but not all. Lemme know if that's ok for you now :)

File: src/components/inputs/PinCodeInput.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nop33) [reply src/components/inputs/PinCodeInput.tsx]: done

File: src/utils/addresses.ts (hunk 1 above)
Comment (nop33) [reply src/utils/addresses.ts]: done

**Review by mvaivre — APPROVED** [review]
Yes! LGTM 👏

**Pull request merged**